src/status_im/hardwallet/core.cljs

```diff
@@ -181,55 +181,51 @@
                 (common/get-application-info pairing nil))
               (when on-verified
-                (dispatch-on-verified-event on-verified)))))
+                (common/dispatch-event on-verified))
+              (clear-on-verify-handlers))))
 
 (fx/defn on-verify-pin-error
   {:events [:hardwallet.callback/on-verify-pin-error]}
   [{:keys [db] :as cofx} error]
-  (let [tag-was-lost? (common/tag-lost? (:error error))
-        setup?        (boolean (get-in db [:hardwallet :setup-step]))
-        exporting?    (get-in db [:hardwallet :on-export-success])]
+  (let [tag-was-lost?       (common/tag-lost? (:error error))
+        setup?              (boolean (get-in db [:hardwallet :setup-step]))
+        on-verified-failure (get-in db [:hardwallet :pin :on-verified-failure])
+        exporting?          (get-in db [:hardwallet :on-export-success])]
     (log/debug "[hardwallet] verify pin error" error)
     (when-not tag-was-lost?
       (if (re-matches common/pin-mismatch-error (:error error))
         (fx/merge cofx
-                  {:db (update-in db [:hardwallet :pin] merge {:status       :error
-                                                               :enter-step   :current
-                                                               :puk          []
-                                                               :current      []
-                                                               :original     []
-                                                               :confirmation []
-                                                               :sign         []
-                                                               :error-label  :t/pin-mismatch})}
+                  {:db (update-in db [:hardwallet :pin]
+                                  merge
+                                  {:status       :error
+                                   :enter-step   :current
+                                   :puk          []
+                                   :current      []
+                                   :original     []
+                                   :confirmation []
+                                   :sign         []
+                                   :error-label  :t/pin-mismatch})}
                   (common/hide-connection-sheet)
-                  (when-not setup?
+                  (when (and (not setup?)
+                             (not on-verified-failure))
                     (if exporting?
                       (navigation/navigate-back)
                       (navigation/navigate-to-cofx :enter-pin-settings nil)))
-                  (common/get-application-info (common/get-pairing db) nil))
+                  (common/get-application-info (common/get-pairing db) nil)
+                  (when on-verified-failure
+                    (fn [_] {:utils/dispatch-later
+                             [{:dispatch [on-verified-failure]
+                               :ms 200}]}))
```

Comment:
Why this number - 200? (Edit - i see it wasn't introduced in this PR, but curious still)

Reply:
it is pretty random, the problem is that we need to wait till bottom sheet will be hidden. But I haven't even tried to put efforts into this as the whole process of account generation (specifically UI) is going to be rewritten soon